situp-core/src/integrationTest/java/com/amazon/situp/integration/EndToEndIntegrationTest.java

```diff
@@ -0,0 +1,194 @@
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.search.SearchHits;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class EndToEndIntegrationTest {
+
+    private static final Random RANDOM = new Random();
+    private static final List<Span.SpanKind> SPAN_KINDS =
+            Arrays.asList(Span.SpanKind.CLIENT, Span.SpanKind.CONSUMER, Span.SpanKind.INTERNAL, Span.SpanKind.PRODUCER, Span.SpanKind.SERVER);
+    private static final String INDEX_NAME = "otel-v1-apm-span-000001";
+
+    @Test
+    public void testPipelineEndToEnd() throws IOException, InterruptedException {
+        //Send data to otel trace source
+        final ExportTraceServiceRequest exportTraceServiceRequest1 = getExportTraceServiceRequest(
+                getRandomResourceSpans(10)
+        );
+
+        final ExportTraceServiceRequest exportTraceServiceRequest2 = getExportTraceServiceRequest(
+                getRandomResourceSpans(10)
+        );
+
+        sendExportTraceServiceRequestToSource(exportTraceServiceRequest1);
+        sendExportTraceServiceRequestToSource(exportTraceServiceRequest2);
+
+        //Wait for data to flow through pipeline and be indexed by ES
+        Thread.sleep(2000);
+
+        //Verify data in elasticsearch sink
+        final List<Map<String, Object>> expectedDocuments = getExpectedDocuments(exportTraceServiceRequest1, exportTraceServiceRequest2);
+        final CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+        credentialsProvider.setCredentials(AuthScope.ANY,
+                new UsernamePasswordCredentials("admin", "admin"));
+        final ConnectionConfiguration.Builder builder = new ConnectionConfiguration.Builder(
+                Collections.singletonList("https://0.0.0.0:9200"));
+        builder.withUsername("admin");
+        builder.withPassword("admin");
+        final RestHighLevelClient restHighLevelClient = builder.build().createClient();
+        final SearchRequest searchRequest = new SearchRequest(INDEX_NAME);
+        searchRequest.source(
+                SearchSourceBuilder.searchSource().size(100)
+        );
+        final SearchResponse searchResponse = restHighLevelClient.search(searchRequest, RequestOptions.DEFAULT);
+        final List<Map<String, Object>> foundSources = getSourcesFromSearchHits(searchResponse.getHits());
+        Assert.assertEquals(expectedDocuments.size(), foundSources.size());
+        expectedDocuments.forEach(expectedDoc -> Assert.assertTrue(foundSources.contains(expectedDoc)));
+    }
+
+    private void sendExportTraceServiceRequestToSource(ExportTraceServiceRequest request) throws InvalidProtocolBufferException {
+        WebClient.of().execute(RequestHeaders.builder()
```

Comment:
I would prefer using gRPC client?

Note: In the longer we should support both gRPC and HTTP/JSON.

Reply:
Sure, I can swap to grpc